### PR TITLE
EVG-8188, EVG-8176, EVG-8188: Configure Patch Page deduplicated tasks & indeterminate checkboxes

### PR DESCRIPTION
**Pull request by SupaJoon**
https://jira.mongodb.org/browse/EVG-7871
https://jira.mongodb.org/browse/EVG-8176
https://jira.mongodb.org/browse/EVG-8188

this PR introduces 

1) mixed (indeterminate) state select all checkbox.
2) deduplicated tasks when multiple build variants are selected. this means that when viewing two build variants at the same time, tasks with the same name will be represented with one checkbox. if the task is selected in one build variant but not the other, the task checkbox will be in an indeterminate state. clicking an indeterminate state task checkbox will uncheck the checkbox. clicking again will check it
3) shift + click allows the user to select multiple build variants

Test the new functionality locally by navigating to: http://localhost:3000/patch/5e6bb9e23066155a993e0f1a/configure/tasks

### CODE REVIEW

File: src/pages/configurePatch/ConfigurePatchCore.tsx [hunk 1]
```diff
@@ -152,19 +152,21 @@ export const ConfigurePatchCore: React.FC<Props> = ({ patch }) => {
...
+        variant: variantName,
+        tasks: tasksArr,
+        displayTasks: [],
+      };
+    })
+    .filter(({ tasks }) => tasks.length);
 
 interface TasksState {
-  [task: string]: true;
+  [task: string]: boolean;
```
[the remark this answers is not in the document]
Comment (SupaJoon) [reply src/pages/configurePatch/ConfigurePatchCore.tsx:169]: yes, that happens here: https://github.com/evergreen-ci/spruce/pull/275/files#diff-b8e749f6d345d275fc7191cafab10303R90

**Comment by SupaJoon** [comment]
I just updated the code to use only one "Select all tasks in this variant" checkbox

File: src/pages/configurePatch/configurePatchCore/ConfigureTasks.tsx [hunk 2]
```diff
@@ -7,6 +7,8 @@ import Checkbox from "@leafygreen-ui/checkbox";
 import { css } from "@emotion/core";
 import isEmpty from "lodash/isEmpty";
 import { Button } from "components/Button";
+import get from "lodash/get";
+import every from "lodash/every";
```
Comment (khelif96) [inline src/pages/configurePatch/configurePatchCore/ConfigureTasks.tsx:11]: I think you forgot to commit the package.json I dont see this package in the package.json. https://github.com/evergreen-ci/spruce/blob/master/package.json#L44

File: src/pages/configurePatch/configurePatchCore/ConfigureBuildVariants.tsx [hunk 3]
```diff
@@ -106,6 +126,7 @@ const BuildVariant = styled.div`
 const VariantName = styled.div`
   word-break: break-all;
   white-space: normal;
+  user-select: none;
```
Comment (khelif96) [inline src/pages/configurePatch/configurePatchCore/ConfigureBuildVariants.tsx:129]: Clever!

**Pull request merged**